responsibleai/responsibleai/rai_insights/rai_insights.py

```diff
@@ -642,6 +625,67 @@ def _validate_rai_insights_input_parameters(
                         'provided has a predict_proba function. '
                         'Please check the task_type.')
 
+        if task_type == ModelTask.CLASSIFICATION:
+            self._validate_classes(
+                model, train, test, target_column, feature_metadata, classes)
+
+    def _validate_classes(
+            self, model, train, test, target_column,
+            feature_metadata, classes):
+        if classes is not None:
+            if (len(set(train[target_column].unique()) -
+                    set(classes)) != 0 or
+                    len(set(classes) -
+                        set(train[target_column].unique())) != 0):
+                raise UserConfigValidationException(
+                    'The train labels and distinct values in '
+                    'target (train data) do not match')
+
+            if (len(set(test[target_column].unique()) -
+                    set(classes)) != 0 or
+                    len(set(classes) -
+                        set(test[target_column].unique())) != 0):
+                raise UserConfigValidationException(
+                    'The train labels and distinct values in '
+                    'target (test data) do not match')
+
+            if model is not None:
+                if feature_metadata is not None and \
+                        feature_metadata.dropped_features is not None and \
+                        len(feature_metadata.dropped_features) != 0:
+                    train_data = train.drop(
+                        columns=feature_metadata.dropped_features + [
+                            target_column],
+                        axis=1)
+                    test_data = test.drop(
+                        columns=feature_metadata.dropped_features + [
+                            target_column],
+                        axis=1)
+                else:
+                    train_data = train.drop(
+                        columns=[target_column], axis=1)
+                    test_data = test.drop(
+                        columns=[target_column], axis=1)
+
+                train_predictions = model.predict(train_data)
+                test_predictions = model.predict(test_data)
+
+                if (len(set(np.unique(train_predictions)) -
+                        set(classes)) != 0 or
+                        len(set(classes) -
+                            set(np.unique(train_predictions))) != 0):
+                    raise UserConfigValidationException(
+                        'The train labels and distinct values in '
+                        'predictions (train data) do not match')
+
+                if (len(set(np.unique(test_predictions)) -
+                        set(classes)) != 0 or
+                        len(set(classes) -
+                            set(np.unique(test_predictions))) != 0):
```

Comment:
nit: since you're repeating this check a couple times, you could make it a helper function that returns a boolean

Reply:
and potentially add the exception in the helper function too